tests/app/questionnaire/test_placeholder_parser.py

```diff
@@ -0,0 +1,464 @@
+            'placeholder': 'persons_name',
+            'transforms': [
+                {
+                    'transform': 'concatenate_list',
+                    'arguments': {
+                        'list_to_concatenate': {
+                            'source': 'answers',
+                            'identifier': ['first-name', 'last-name']
+                        },
+                        'delimiter': ' '
+                    }
+                },
+                {
+                    'transform': 'format_possessive',
+                    'arguments': {
+                        'string_to_format': {
+                            'source': 'previous_transform'
+                        }
+                    }
+                }
+            ]
+        }]
+
+        parser = PlaceholderParser(answer_store=AnswerStore([
+            {
+                'answer_id': 'first-name',
+                'value': 'Joe'
+            },
+            {
+                'answer_id': 'last-name',
+                'value': 'Bloggs'
+            }
+        ]))
+
+        placeholders = parser.parse(placeholder_list)
+
+        assert placeholders['persons_name'] == 'Joe Bloggs’'
+
+    @staticmethod
+    def test_list_transform_placeholder():
+        placeholder_list = [{
+            'placeholder': 'full_name_list',
+            'transforms': [
+                {
+                    'transform': 'filter_list',
+                    'arguments': {
+                        'source_list': {
+                            'source': 'list',
+                            'identifier': 'people'
+                        },
+                        'answer_ids': ['first-name', 'last-name'],
+                        'concatenate_with': ' '
+                    }
+                }
+            ]
+        }]
+
+        parser = PlaceholderParser(lists={
+            'people': [
+                {
+                    'answer_id': 'first-name',
+                    'group_id': 'name-group',
+                    'group_instance': 0,
+                    'value': 'Joe'
+                },
+                {
+                    'answer_id': 'last-name',
+                    'group_id': 'name-group',
+                    'group_instance': 0,
+                    'value': 'Bloggs'
+                },
+                {
+                    'answer_id': 'first-name',
+                    'group_id': 'name-group',
+                    'group_instance': 1,
+                    'value': 'Jane'
+                },
+                {
+                    'answer_id': 'last-name',
+                    'group_id': 'name-group',
+                    'group_instance': 1,
+                    'value': 'Bloggs'
+                }
+            ]
+        })
+
+        placeholders = parser.parse(placeholder_list)
+
+        assert placeholders['full_name_list'] == ['Joe Bloggs', 'Jane Bloggs']
+
+    @staticmethod
+    def test_list_transform_placeholder_no_concatenate():
```

Comment:
I think we should remove the list functionality from this pull request. The structure of the list used in the tests doesn't match the proposed structure, and that could also change when we implement lists.

Reply:
Sure, I've removed this behaviour now.